src/packaging/kpatch/commit_to_overlay.sh

```diff
@@ -0,0 +1,459 @@
+  echo "Checking out the change to build and post it"
+  git checkout ${change}
+
+  # Capture the commit message from the change
+  echo "Creating the initial commit file using the commit message from the change"
+  commit_file=$(mktemp /tmp/overlay_commit_file.XXXXX)
+  git log --format=%B -n 1 ${change} > ${commit_file}
+
+  # Capture the source commit hash into the commit message as well
+  echo "Original commit: ${change}" >> ${commit_file}
+
+  # Capture authorship information for use later
+  commit_author_email=$(git log --format=%aE -1 ${change})
+  commit_author_name=$(git log --format=%aN -1 ${change})
+  commit_date=$(git log --format=%ad -1 ${change})
+  gitauthor="--author \"${commit_author_name} <${commit_author_email}>\""
+  gitdate="--date \"${commit_date}\""
+
+  # Build the tree as normal
+  build_log_file=$(mktemp /tmp/overlay_build_log.XXXXX)
+  echo -en "Building the VDO tree... "
+  make >> ${build_log_file} 2>&1
+  echo "Done"
+
+  if [[ $? != 0 ]]; then
+    echo -e "${COLOR_RED}ERROR: Building the VDO tree failed. See ${build_log_file} for more" \
+            "information.${NO_COLOR}"
+    exit
+  fi
+
+  # Generate the kernel overlay and apply it to the linux tree
+  echo "Generating the kernel overlay"
+  cd src/packaging/kpatch
+  cp -p Makefile Makefile.orig
+
+  sed -r -i -e 's/^GITARGS=/GITARGS=\nGITAUTHOR ?= ""/g'        \
+            -e 's/^GITARGS=/GITARGS=\nGITDATE ?= ""/g'          \
+            -e 's/commit -m/commit --file/g'                     \
+            -e 's/ commit / commit $(GITAUTHOR) $(GITDATE) /g'  \
+            -e 's/^(\s*&&).*format-patch.*/\1 echo committed/g' \
+            -e "s/ origin\/master / origin\/${KERNEL_BRANCH} /g" Makefile
+
+  export GITDATE="${gitdate}" GITAUTHOR="${gitauthor}" CHANGE_LOG="${commit_file}" \
+         LINUX_SRC=${LINUX_SRC}
+  make prepare overlay >> ${build_log_file}
+
+  if [[ $? != 0 ]]; then
+    echo -e "${COLOR_RED}ERROR: Generating the kernel overlay failed. See ${build_log_file} for" \
+            "more information.${NO_COLOR}"
+    exit
+  fi
+
+  echo "Applying the kernel overlay to branch '${OVERLAY_BRANCH}'"
+  make kpatch >> ${build_log_file} 2>&1
+
+  if [[ $? != 0 ]]; then
+    return=$(tail -5 ${build_log_file} | grep "^[nN]othing to commit" | wc -l)
+
+    if [[ $return > 0 ]]; then
+      echo "Nothing to apply - moving on."
+    else
+      echo -e "${COLOR_RED}ERROR: Applying the kernel overlay to branch '${OVERLAY_BRANCH}'" \
+              "failed. See ${build_log_file} for more information.${NO_COLOR}"
+      exit
+    fi
+  fi
+
+  # Clean up
+  echo "Cleaning up build artifacts"
+  mv Makefile.orig Makefile
+
+  rm -fv ${commit_file}
+
+  if [[ ${DEBUG} != 0 ]]; then
+    rm -fv ${build_log_file}
+  fi
+
+  unset -f GITDATE GITAUTHOR CHANGE_LOG
+done
+
+# Push the kernel overlay branch to the remote kernel repository
+cd ${LINUX_SRC}
+prompt_user_push
+
+if [[ ${MANUAL_PUSH} != 0 ]]; then
+  git push -u origin ${OVERLAY_BRANCH}
```

Comment:
I feel like the MANUAL_PUSH flag is set backwards. Wouldn't MANUAL_PUSH =1 mean leave the push to the user, and MANUAL_PUSH=0 mean do the push automatically?

Reply:
For shell scripts, true is 0 and 1 is false. As such, I was keeping with that theme so MANUAL_PUSH true is 0, automatic push is MANUAL_PUSH false, which is 1.

Reply:
I ... can see the logic. Seems weird to me but I wouldn't want to make the style non-standard in a confusing way.